Python/immutability.c

```diff
@@ -156,281 +239,877 @@ static bool is_c_wrapper(PyObject* obj){
     return PyCFunction_Check(obj) || Py_IS_TYPE(obj, &_PyMethodWrapper_Type) || Py_IS_TYPE(obj, &PyWrapperDescr_Type);
 }
 
-// Lifted from Python/gc.c
-//******************************** */
-#ifndef Py_GIL_DISABLED
-#define GC_NEXT _PyGCHead_NEXT
-#define GC_PREV _PyGCHead_PREV
+static inline void _Py_SetImmutable(PyObject *op)
+{
+    if(op) {
+#if SIZEOF_VOID_P > 4
```

Comment:
Suggestion: Should we define one macro like `_PY_IMMUTABLE_STORE_IN_RC` or something to indicate where we store the flag instead of using `SIZEOF_VOID_P` everywhere? Then we could also add documentation to the macro explaining why we store it when where

Reply:
There are macros for which bits, and how to mask, so I think it worked out well.  Thanks for highlighting it wasn't great.